src/Microsoft.AspNet.Mvc.Core/HttpResponseStreamWriter.cs

```diff
@@ -0,0 +1,244 @@
+        {
+            if (values == null)
+            {
+                return;
+            }
+
+            while (count > 0)
+            {
+                if (_charBufferCount == _charBufferSize)
+                {
+                    FlushInternal();
+                }
+
+                CopyToCharBuffer(values, ref index, ref count);
+            }
+        }
+
+        public override void Write(string value)
+        {
+            if (value == null)
+            {
+                return;
+            }
+
+            var count = value.Length;
+            var index = 0;
+            while (count > 0)
+            {
+                if (_charBufferCount == _charBufferSize)
+                {
+                    FlushInternal();
+                }
+
+                CopyToCharBuffer(value, ref index, ref count);
+            }
+        }
+
+        public override async Task WriteAsync(char value)
+        {
+            if (_charBufferCount == _charBufferSize)
+            {
+                await FlushInternalAsync();
+            }
+
+            _charBuffer[_charBufferCount] = value;
+            _charBufferCount++;
+        }
+
+        public override async Task WriteAsync(char[] values, int index, int count)
+        {
+            if (values == null)
+            {
+                return;
+            }
+
+            while (count > 0)
+            {
+                if (_charBufferCount == _charBufferSize)
+                {
+                    await FlushInternalAsync();
+                }
+
+                CopyToCharBuffer(values, ref index, ref count);
+            }
+        }
+
+        public override async Task WriteAsync(string value)
+        {
+            if (value == null)
+            {
+                return;
+            }
+
+            var count = value.Length;
+            var index = 0;
+            while (count > 0)
+            {
+                if (_charBufferCount == _charBufferSize)
+                {
+                    await FlushInternalAsync();
+                }
+
+                CopyToCharBuffer(value, ref index, ref count);
+            }
+        }
+
+        // We want to flush the stream when Flush/FlushAsync is explicitly
+        // called by the user (example: from a Razor view).
+
+        public override void Flush()
+        {
+            FlushInternal(true, true);
+        }
+
+        public override async Task FlushAsync()
+        {
+            await FlushInternalAsync(true, true);
+        }
+
+        // Do not flush the stream on Close/Dispose, as this will cause response to be
+        // sent in chunked encoding in case of Helios.
+
+#if DNX451
+        public override void Close()
+        {
+            FlushInternal(flushStream: false, flushEncoder: true);
+        }
+#endif
+        protected override void Dispose(bool disposing)
+        {
+            // In CoreClr this is equivalent to calling Close()
```

Comment:
:+1: ...in full .net call on `Close` in turn calls `Dispose`, so I can remove this...